source/common/access_log/access_log_impl.cc

```diff
@@ -77,6 +77,13 @@ FilterFactory::fromProto(const envoy::config::filter::accesslog::v2::AccessLogFi
   case envoy::config::filter::accesslog::v2::AccessLogFilter::kGrpcStatusFilter:
     MessageUtil::validate(config);
     return FilterPtr{new GrpcStatusFilter(config.grpc_status_filter())};
+  case envoy::config::filter::accesslog::v2::AccessLogFilter::kExtensionFilter:
+    MessageUtil::validate(config);
+    {
+      auto& factory = Config::Utility::getAndCheckFactory<ExtensionFilterFactory>(
+          config.extension_filter().name());
+      return factory.createFilter(config.extension_filter(), runtime, random);
+    }
```

Comment:
Yeah, I would love to get rid of the explicit switch, but...

... I've tried to get existing filters into the factory registration mechanism, but couldn't get there without loosing filter-specific config validation or changes to existing filter API. 

I've discussed this with @htuch and confirmed that adding 'ExtensionFilter' is a pattern similar to other places, so we'll have to stay with that.

Reply:
Yeah, I think for terseness of config it might be better not to make all of these extensions, but I do definitely concede the point that @auni53 makes that it would be simpler in implementation to have everything dynamic.

Reply:
Ack, thanks for confirmation!